utils/das_data_sorter.py

```diff
@@ -1,6 +1,79 @@
 import pandas as pd 
 import argparse 
 from math import ceil
+from numpy import median, mean
+
+# averages data within the seconds or minutes, according to new_time array
+def avg_Data(array):
+    avg_array = []
+
+    for millisecond_indices in new_time:
+        avg = mean(array[millisecond_indices])
+        avg = round(avg, ndigits=2)
+        avg_array.append(avg)
+
+    return avg_array
+
+# 3 point moving mean smoothing
+def mean_smooth(data):
+    new_data = []
+
+    for i in range(1,len(data)-1):
+        first = data[i-1]
+        second = data[i]
+        third = data[i+1]
+
+        new_data_point = mean([first, second, third])
+        new_data_point = round(new_data_point, ndigits=2)
+        new_data.append(new_data_point)
+    
+    return new_data
+
+# 3 point moving median smoothing
+def median_smooth(data):
+    new_data = []
+
+    # iterates through each data point and its adjacents (excluding first and last) and finds the median 
+    for i in range(1, len(data)-1):
+        first = data[i-1]
+        second = data[i]
+        third = data[i+1]
+
+        new_data_point = median([first, second, third])
+        new_data.append(new_data_point)
+    
+    return new_data
+
+# smoothing technique function caller
+def smooth_data(data, technique):
+    if technique == "mean":
```

Comment:
The if statements are quite unnecessarily long. When I look at the two if statements, I can see that they are very similar. The only difference is that we use a different calculation method depending on the technique. Try assigning the technique function to a variable so it is cleaner.

Here is an example of how I would resolve this problem.

``` python
smoothing_method = None
if technique is "mean":
	smoothing_method = mean_smooth
elif technique is "median":
	smoothing_method = median_smooth
else:
	raise ValueError("Invalid technique given")

final_data = {
            "time": range(1, len(new_time)-1),
            "gps_course": smoothing_method(data["gps_course"]),
            "gps_speed": smoothing_method(data["gps_speed"]),
            "tempC": smoothing_method(data["tempC"]),
            "tempF": smoothing_method(data["tempF"]),
            "cadence": smoothing_method(data["cadence"]),
            "power": smoothing_method(data["power"]),
            "reed_velocity": smoothing_method(data["reed_velocity"]),
            "reed_distance": smoothing_method(data["reed_distance"])
        }
return final_data
```

Reply:
In general when coding, if you've ever had to type the same thing twice, there's almost always a better way :wink: